backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/floatingnotification/BpkFloatingNotification.kt

```diff
@@ -0,0 +1,148 @@
+import androidx.compose.animation.core.FiniteAnimationSpec
+import androidx.compose.animation.core.LinearEasing
+import androidx.compose.animation.core.tween
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.animation.slideInVertically
+import androidx.compose.animation.slideOutVertically
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.requiredHeightIn
+import androidx.compose.foundation.layout.requiredSize
+import androidx.compose.foundation.layout.requiredWidthIn
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.delay
+import net.skyscanner.backpack.compose.card.BpkCard
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.icon.BpkIconSize
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.utils.clickable
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.DurationUnit
+
+@Composable
+fun BpkFloatingNotification(
+  modifier: Modifier = Modifier,
+  text: String,
+  show: Boolean,
+  icon: BpkIcon? = null,
+  animation: Animation = Animation(),
+  cta: CTA? = null
+) {
+  val (internalShow, setInternalShow) = remember { mutableStateOf(false) }
+  if (show) {
+    LaunchedEffect(key1 = Unit) {
+      setInternalShow(true)
+      delay(animation.showDuration)
+      setInternalShow(false)
+      animation.onFinished?.invoke()
+    }
+  }
+  val animationDuration = if (animation.enabled) animation.transitionDuration.toInt(DurationUnit.MILLISECONDS) else 0
+  val fadeAnimationSpec: FiniteAnimationSpec<Float> = tween(durationMillis = animationDuration, easing = LinearEasing)
+  val slideAnimationSpec: FiniteAnimationSpec<IntOffset> = tween(durationMillis = animationDuration, easing = LinearEasing)
+  AnimatedVisibility(
+    visible = internalShow,
+    enter = slideInVertically(slideAnimationSpec, initialOffsetY = { it / 2 }) + fadeIn(animationSpec = fadeAnimationSpec),
+    exit = slideOutVertically(slideAnimationSpec, targetOffsetY = { it / 2 }) + fadeOut(animationSpec = fadeAnimationSpec)
+  ) {
+    Box(
+      modifier = modifier
```

Comment:
Modifier param should be used only once, for most of the cases – for top-level layouts